ppb/systems.py

```diff
@@ -70,48 +80,27 @@ def mouse_motion(self, event, scene):
         buttons = [bool(x) for x in event.buttons]
         return events.MouseMotion(game_position, screen_position, delta, buttons)
 
-class MouseSystem(System):
-    """
-    Mouse system.
-    """
-
-    def __init__(self, *, engine, resolution=default_resolution, **kwargs):
-        self.mouse = Mouse()
-        self.offset = Vector(-0.5 * resolution[0],
-                             -0.5 * resolution[1])
-        engine.register(events.Update, "mouse", self.mouse)
-
-    def activate(self, engine):
-        """Sync mouse with hardware."""
-        buttons = self.get_hardware_buttons()
-        if buttons is not None:
-            self.mouse.buttons = [bool(x) for x in buttons]
-        screen_position = self.get_hardware_position()
-        if screen_position is not None:
-            camera = engine.current_scene.main_camera
-            self.mouse.screen_position = Vector(*screen_position)
-            self.mouse.position = camera.translate_to_frame(self.mouse.screen_position)
-        return []
-
-    def on_mouse_motion(self, mouse_motion_event, signal):
-        self.mouse.buttons = mouse_motion_event.buttons
-        self.mouse.screen_position = mouse_motion_event.screen_position
-        self.mouse.position = mouse_motion_event.position
-
-    def get_hardware_buttons(self) -> Union[Iterable, None]:
-        pass
-
-    def get_hardware_position(self) -> Union[Iterable, None]:
-        pass
-
-
-class PygameMouseSystem(MouseSystem):
-
-    def get_hardware_buttons(self):
-        return pygame.mouse.get_pressed()
-
-    def get_hardware_position(self):
-        return pygame.mouse.get_pos()
+    def button_pressed(self, event, scene):
+        screen_position = Vector(*event.pos)
+        camera = scene.main_camera
+        game_position = camera.translate_to_frame(screen_position)
+        btn = self.button_map.get(event.button)
+        if btn is not None:
+            return events.ButtonPressed(
+                button=btn,
+                position=game_position,
+            )
```

Comment:
We definitely need screen space on these.

Reply:
1. Why?
2. This isn't available on pyglet